source/slang/hlsl.meta.slang

```diff
@@ -9571,6 +9571,24 @@ void CallShader(uint shaderIndex, inout Payload payload)
 
 // 10.3.2
 
+// This intrinsic returns `maybeStruct` if a struct. If `maybeStruct` is a non struct it returns a `struct{T maybeStruct};` object
+// the intrinsic will copy `struct{T maybeStruct};` into `maybeStruct` if the parameter can exhibit side effects (`inout`, `out`, `ref`, etc.)
```

Comment:
This comment should be more of "why" we need this function more than what it does.